### PR TITLE
[CLI] Remove some macros

### PR DESCRIPTION
**Pull request by serban300**
Refactor the implementations for the following CLI methods in order to
avoid using macros:
- init_bridge
- relay_headers
- relay_parachains

This is an initial PR to gather feedback on the idea of using traits instead of macros for implementing the CLI methods. In my opinion this would bring 2 benefits:
- better code readability
- some code deduplication

If there are no concerns related to this approach we can move on and try to use it for the other commands as well. The final purpose is to try to use this for `relay_headers_and_messages`. If this works I think we can address #1415 more naturally, probably also using traits.

### CODE REVIEW

File: relays/bin-substrate/src/cli/init_bridge.rs [hunk 1]
```diff
@@ -52,188 +61,42 @@ pub enum InitBridgeName {
...
+
+	/// Get the encoded call to init the bridge.
+	fn encode_init_bridge(
+		init_data: <Self::Engine as Engine<Self::Source>>::InitializationData,
+	) -> <Self::Target as Chain>::Call;
+
+	/// Initialize the bridge.
+	fn init_bridge(
+		data: InitBridge,
+	) -> Pin<Box<dyn Future<Output = anyhow::Result<()>> + Send + 'static>> {
```
Comment (svyatonik) [inline relays/bin-substrate/src/cli/init_bridge.rs:78]: Imo it would be better to use `async_trait` here instead of all this `Pin<Box<...>>` stuff :)

Comment (svyatonik) [reply relays/bin-substrate/src/cli/init_bridge.rs:78]: (there's also couple of other places where imo it shall be replaced)

File: relays/bin-substrate/src/cli/bridge.rs [hunk 2]
```diff
@@ -218,3 +221,137 @@ macro_rules! select_full_bridge {
...
+//// `RialtoParachain` to `Millau` bridge definition.
+pub struct RialtoParachainToMillauCliBridge {}
+
+impl CliBridgeBase for RialtoParachainToMillauCliBridge {
+	type Source = relay_rialto_parachain_client::RialtoParachain;
+	type Target = relay_millau_client::Millau;
+}
+
+//// `WestendParachain` to `Millau` bridge definition.
+pub struct WestendParachainToMillauCliBridge {}
```
Comment (svyatonik) [inline relays/bin-substrate/src/cli/bridge.rs:352]: Let's rename it to `WestmintToMillauCliBridge` maybe? There are many parachains on Westend and iiuc this struct is dedicated to Westmint.

**Review by svyatonik — APPROVED** [review]
Changes seems good to me :) I hope you have a vision for the final solution, which will actually make the code easier to read :) Couple of suggestions.

File: relays/bin-substrate/src/cli/bridge.rs (hunk 2 above)
Comment (serban300) [reply relays/bin-substrate/src/cli/bridge.rs]: Done.

File: relays/bin-substrate/src/cli/init_bridge.rs (hunk 1 above)
Comment (serban300) [reply relays/bin-substrate/src/cli/init_bridge.rs]: Indeed, it looks better. Done.

**Review by svyatonik — APPROVED** [review]
Cool, thank you!

**Pull request merged**